### PR TITLE
Feature/admin events

### PR DESCRIPTION
**Pull request by Sagensagen**
# :warning: [Dependent on tdctl_api/feature/participants](https://github.com/td-org-uit-no/tdctl-api/pull/41) :warning: 
# Admin events ++ 
 In order to conduct admin event i had to restructure events and participants which lead to adapt a lot of features, resulting in this quite huge PR.
## Admin event page
Put all event handling to a single page. 
### :gear:  Settings
- Moved event settings here exclusively
- Added active/inactive toggle to settings
### Responses
Table of all members that has submitted as participants. 
Can Remove participants through table interface
image

> Table might  be incorrect due to seeding data creating identical submitDates and includes multiple participants exceeding  maxParticipants. Should not be a problem in production. Can be fixed by seeding script.
TODO: food/Transport is hard coded into the form. Needs to be added to be selected. @otytlandsvik Is on this one i think?

### :bar_chart:  Statistics
General statistics for respective event
- How many needs food
- How many have submitted from total TD members
- Class distribution
- How many need Transport
- Participant submission X time chart
- Dietary restrictions overview
image

> This page is computing a lot of stuff which might could have been solved by  the backend and additional api's. I have not measured any overhead, but it is not critical and works fine as is. Time chart is not scaling depending on inputs, and  can be optimized in future. 

### :page_facing_up: Export event details if  event data needed in sheet format. 
# :purple_square: Eventbody
Added a modal+form to appear if event requires participant to select food/transport/dietary restrictions.
Added event/admin navigation button for admins
image

# :classical_building:  Event & participant model
Changed both models to fit api
# Other
## Toggle molecule
Changed color to  purple/theme color for theme consistency
Added initial value to prefix its state when binding
## Icon 
Extended props to take optional color
## api
Added deleteParticipant api

### CODE REVIEW

**Review by FredrikMorstad — CHANGES_REQUESTED** [review]
Left some comments, that should to be fixed

File: package.json [hunk 1]
```diff
@@ -58,6 +59,7 @@
     ]
   },
   "devDependencies": {
-    "@iconify/react": "^3.0.0"
+    "@iconify/react": "^3.0.0",
+    "sass": "^1.55.0"
```
[the remark this answers is not in the document]
Comment (Sagensagen) [reply package.json:63]: @FredrikMorstad This was inherited from master branch. Should be moved to "dependencies"?

File: src/components/atoms/icons/icon.tsx [hunk 2]
```diff
@@ -3,13 +3,15 @@ import React from 'react';
 interface Props extends React.HTMLAttributes<HTMLSpanElement> {
   type: string;
   size?: number;
+  color?: string;
 }
 
-const Icon = ({ type, size = 1, onClick }: Props) => (
+const Icon = ({ type, size = 1, color = 'primary', onClick }: Props) => (
```
[the remark this answers is not in the document]
Comment (Sagensagen) [reply src/components/atoms/icons/icon.tsx:9]: `primary` does not trigger anything and base color is white. Can be misinterpreted, change to white.

File: src/components/molecules/event/eventExport/eventExport.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+import { baseUrl } from 'constants/apiConstants';
+import { getTokens } from 'utils/auth';
+
+const EventExport: React.FC<{ event: Event }> = ({ event }) => {
+  const filename = event.title + '.xlsx';
+
+  const exportEventCsv = async (url: string) => {
+    const { accessToken } = getTokens();
+    const request = new Request(baseUrl + 'event/' + url + '/export', {
```
[the remark this answers is not in the document]
Comment (Sagensagen) [reply src/components/molecules/event/eventExport/eventExport.tsx:13]: Initially i tried that without success. Leeched of the get_image request to make it work.

File: src/components/molecules/event/eventButton/EventButton.tsx [hunk 4]
```diff
@@ -27,16 +29,51 @@ const AuthEventButton: React.FC<AuthButtonProps> = ({
...
+        case 400:
+          console.log(400);
+          break;
+        case 404:
+          console.log(404);
+          break;
+        default:
+          console.log(error.statusCode);
+      }
+    }
```
[the remark this answers is not in the document]
Comment (Sagensagen) [reply src/components/molecules/event/eventButton/EventButton.tsx:69]: Yes, due to the buttonactions are "booelean" in terms of they cannot handle conditions. This was necessary to verify if join request needed any payload.

File: src/components/molecules/event/eventExport/eventExport.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (FredrikMorstad) [reply src/components/molecules/event/eventExport/eventExport.tsx:13]: jeep, I did the same as well:mag: I think we can add a comment for later restructure :label:

File: src/components/molecules/event/eventButton/EventButton.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (FredrikMorstad) [reply src/components/molecules/event/eventButton/EventButton.tsx:69]: make sense :+1:

File: src/components/pages/admin/AdminPage.tsx [hunk 5]
```diff
@@ -1,6 +1,7 @@
 import React, { useState } from 'react';
 import styles from './adminPage.module.scss';
 import { MemberTable } from 'components/molecules/tables';
+import { EventTable } from 'components/molecules/tables';
```
Comment (FredrikMorstad) [inline src/components/pages/admin/AdminPage.tsx:4]: Can be imported from same file

**Review by FredrikMorstad — CHANGES_REQUESTED** [review]
Some small changes that needs to be added otherwise LGTM :+1:

File: src/components/molecules/event/eventStatistics/EventStatistics.tsx [hunk 6]
```diff
@@ -0,0 +1,436 @@
...
+
+  // Groups an array of objects by a given key into an object of the keys
+  function groupBy<T>(collection: T[], key: keyof T) {
+    const groupedResult = collection.reduce((previous, current) => {
+      if (!previous[current[key]]) {
+        previous[current[key]] = [] as T[];
+      }
+      previous[current[key]].push(current);
+      return previous;
+    }, {} as any);
```
[the remark this answers is not in the document]
Comment (Sagensagen) [reply src/components/molecules/event/eventStatistics/EventStatistics.tsx:376]: nope :-1:

**Review by FredrikMorstad — APPROVED** [review]
LGTM :rocket:

**Pull request merged**